library/packages/src/lib/y2packager/product_control_product.rb

```diff
@@ -0,0 +1,102 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE.
+#
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "y2packager/product_license_mixin"
+
+Yast.import "Arch"
+Yast.import "Linuxrc"
+Yast.import "ProductFeatures"
+
+module Y2Packager
+  # This class implements a base product read from the control.xml file.
+  class ProductControlProduct
+    # initialize the selected base product
+    @selected = nil
+
+    class << self
+      attr_accessor :selected
+    end
+
+    extend Yast::Logger
+    include ProductLicenseMixin
+
+    attr_reader :name, :version, :arch, :label, :license_url, :register_target
+
+    #
+    # Read the base products from the control.xml file. The products for the incompatible
+    # machine architecture and the hidden products are filtered out.
+    #
+    # @return [Array<Installation::ProductControlProduct>] List of the products
+    def self.products
```

Comment:
why not having this code in above `self << class` ?

Reply:
That was actually added/changed by Rubocop, I'll add it there as well.